File: mobile/application/app/src/main/java/com/example/choiceiomobile/data/models/FavouriteDto.kt
package com.example.choiceiomobile.data.models

// Для POST и DELETE запросов
data class FavouriteOperationResponseDto(
    val success: Boolean,
    val message: String,
    val animeId: Int,
    val userId: Int
)

// Для GET запроса
data class FavouritesListResponseDto(
    val success: Boolean,
    val userId: Int,
    val count: Int,
    val animeIds: List<Int>
)

// Для POST и DELETE запросов (тело запроса)
data class FavouriteRequestDto(
    val animeId: Int,
    val userId: Int
)

File: backend/api-service/Api/Api/Models/DTOs/Favourite/FavouriteDto.cs
namespace Api.Models
{
    public class FavouriteDto
    {
        public long AnimeId { get; set; }
        public int UserId { get; set; }
    }
    
    public class AddFavouriteRequest
    {
        public long AnimeId { get; set; }
        public int UserId { get; set; }
    }
    
    public class RemoveFavouriteRequest
    {
        public long AnimeId { get; set; }
        public int UserId { get; set; }
    }
}